File: Arrays/2733. Neither Minimum nor Maximum/2733. Neither Minimum nor Maximum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findNonMinOrMax = function (nums) {
  if (nums.length <= 2) return -1;

  const minn = Math.min(...nums);
  const maxx = Math.max(...nums);

  for (const n of nums) {
    if (n !== minn && n !== maxx) return n;
  }
};

// Time: O(n)
// Space: O(1)

File: Arrays/2733. Neither Minimum nor Maximum/2733. Neither Minimum nor Maximum.py
class Solution:
    def findNonMinOrMax(self, nums: List[int]) -> int:
        if len(nums) <= 2:
            return -1
        
        minn = min(nums)
        maxx = max(nums)

        for n in nums:
            if n != minn and n != maxx:
                return n

# Time: O(n)
# Space: O(1)